### PR TITLE
PTL create_moms() method creates moms if pbsnodes list is empty

### PR DESCRIPTION
**Pull request by birbalsain**
#### Describe Bug or Feature
PTL create_moms() method failed to create moms if pbsnodes list is empty.  This method  first delete all the existing nodes and checks the status of pbsnodes to confirm all nodes are deleted. After that it creates moms.
If pbsnodes list is empty, This method throws  PbsStatusError "Server has no node list" while checking the status of pbsnodes.

#### Describe Your Change
It accepts  "PbsStatusError" error "Server has no node list" because it is expected  while checking the empty pbsnodes list.

#### Attach Test and Valgrind Logs/Output

before changes

[pbs_config_create_mom_before.txt](https://github.com/openpbs/openpbs/files/7049329/pbs_config_create_mom_before.txt)

After changes
[pbs_config_create_mom.txt](https://github.com/openpbs/openpbs/files/7049333/pbs_config_create_mom.txt)

### CODE REVIEW

**Review by visheshh — APPROVED** [review]
LGTM

**Comment by kjakkali** [comment]
@openpbs/openpbs-maintainers  Please take this forward.

File: test/fw/ptl/lib/ptl_server.py [hunk 1]
```diff
@@ -1860,7 +1860,14 @@ def create_moms(self, name=None, attrib=None, num=1, delall=True,
             except PbsManagerError as e:
                 rc = e.rc
             if rc:
-                if len(self.status(NODE)) > 0:
+                node_length = 0
+                try:
+                    node_length = len(self.status(NODE))
+                except PbsStatusError as err:
+                    if "Server has no node list" not in err.msg[0]:
+                        self.logger.error("node sttus err is " + str(err))
```
Comment (bhroam) [inline test/fw/ptl/lib/ptl_server.py:1868]: This isn't a very informative error message.  Could you try to make it a little more explanatory to what the problem is?

File: test/fw/ptl/lib/ptl_server.py [hunk 2]
```diff
@@ -1860,7 +1860,14 @@ def create_moms(self, name=None, attrib=None, num=1, delall=True,
             except PbsManagerError as e:
                 rc = e.rc
             if rc:
-                if len(self.status(NODE)) > 0:
+                node_length = 0
+                try:
+                    node_length = len(self.status(NODE))
```
Comment (bhroam) [inline test/fw/ptl/lib/ptl_server.py:1865]: If pbsnodes doesn't return any nodes, shouldn't status() return [] and not an exception?  Not having any nodes (or queues for that matter) is a valid PBS configuration.

**Review by bhroam — CHANGES_REQUESTED** [review]

File: test/fw/ptl/lib/ptl_server.py (hunk 2 above)
Comment (birbalsain) [reply test/fw/ptl/lib/ptl_server.py:1865]: If node is exist but PTL method "create_moms()" creates moms after deleting all the existing nodes and checking nodes are deleted. If a node doesn't exist then pbs command "/opt/pbs/bin/pbsnodes -s centos.pbspro.com -v -a"  itself through a error. err: ['/opt/pbs/bin/pbsnodes: Server has no node list']

**Review by bhroam — APPROVED** [review]

**Pull request merged**